sources/dfmc/reader/tests/literal-test-suite.dylan

```diff
@@ -0,0 +1,83 @@
+Module: dfmc-reader-test-suite
+License: See License.txt in this distribution for details.
+
+define function verify-literal
+    (f :: <fragment>, value, required-class)
+ => ()
+  assert-equal(f.fragment-value, value);
+  assert-true(instance?(f, required-class));
+end function verify-literal;
+
+define function verify-presentation
+    (f :: <fragment>, presentation :: <string>)
+ => ()
+  let stream = make(<string-stream>, direction: #"output");
+  present-fragments(list(f), stream);
+  assert-equal(stream.stream-contents, presentation);
+end function verify-presentation;
+
+define test boolean-literal-test ()
+  let t = read-fragment("#t");
+  verify-literal(t, #t, <true-fragment>);
+  verify-presentation(t, "#t");
+
+  let f = read-fragment("#f");
+  verify-literal(f, #f, <false-fragment>);
+  verify-presentation(f, "#f");
+end test boolean-literal-test;
+
+define test character-literal-test ()
+  let f = read-fragment("'n'");
+  verify-literal(f, 'n', <character-fragment>);
+  verify-presentation(f, "'n'");
+
+  let f = read-fragment("'N'");
+  assert-equal(f.fragment-value, 'N');
+  verify-presentation(f, "'N'");
+
+  let f = read-fragment("'\\\\'");
+  assert-equal(f.fragment-value, '\\');
+  verify-presentation(f, "'\\\\'");
+
+  let f = read-fragment("'\\n'");
+  assert-equal(f.fragment-value, '\n');
+  verify-presentation(f, "'\\n'");
+
+  let f = read-fragment("'\\<01>'");
+  assert-equal(as(<integer>, f.fragment-value), 1);
+  verify-presentation(f, "'\\<1>'");
```

Comment:
This seems to test basic \<...> syntax and handling of leading zeros.  How about a test to verify it's handling only hex digits?